src/librustc_codegen_llvm/debuginfo/metadata.rs

```diff
@@ -1504,7 +1507,8 @@ impl VariantMemberDescriptionFactory<'ll, 'tcx> {
 
 #[derive(Copy, Clone)]
 enum EnumDiscriminantInfo<'ll> {
-    RegularDiscriminant(&'ll DIType),
+    /// First field is the field index of the discriminant.
+    RegularDiscriminant(usize, &'ll DIType),
```

Comment:
Use https://github.com/rust-lang/rust/blob/1d4f0e71c6cd55d43f1d7b9b58cab8ef7c6f4117/src/librustc/mir/mod.rs#L2004 as the type instead of `usize` and adjust all use sites to match if possible (instead of using `usize` or `u32`). The latter can also be a separate PR, since it's a preexisting issue. You can use the `as_u32` and `as_usize` methods on `Field` for now.

Reply:
Done.